src/main/org/deidentifier/arx/ARXConfiguration.java

```diff
@@ -707,7 +707,9 @@ public Monotonicity getMonotonicityOfPrivacy() {
      * @return
      */
     public Monotonicity getMonotonicityOfUtility() {
-        if (metric.isMonotonic() || (this.getMaxOutliers() == 0d) || this.isPracticalMonotonicity()) {
+        if ((metric.isMonotonicWithGeneralization() && this.getMaxOutliers() == 0) || 
```

Comment:
Ok, great. I think that this bug is fixed. However, we don't have a unit test for the Stackelberg game that benefits from the new code path (assuming monotonicity without it actually being present does not necessarily lead to wrong results and this fix only affects configurations with a suppression limit of 0%). Have you checked that ARX is now executing the correct algorithm?